source/Calamari/Kubernetes/Commands/KubernetesApplyRawYamlCommand.cs

```diff
@@ -20,109 +22,124 @@
-                    if (fileSystem.FileExists(inlineFile))
+                    if (pathToPackage != null)
+                    {
+                        extractPackage.ExtractToStagingDirectory(pathToPackage);
+                    }
+                    else
                     {
-                        fileSystem.CopyFile(inlineFile, Path.Combine(stagingDirectory, "customresource.yml"));
+                        //If using the inline yaml, copy it to the staging directory.
+                        var inlineFile = Path.Combine(d.CurrentDirectory, "customresource.yml");
+                        var stagingDirectory = Path.Combine(d.CurrentDirectory, "staging");
+                        fileSystem.EnsureDirectoryExists(stagingDirectory);
+                        if (fileSystem.FileExists(inlineFile))
+                        {
+                            fileSystem.MoveFile(inlineFile, Path.Combine(stagingDirectory, "customresource.yml"));
+                        }
+
+                        d.StagingDirectory = stagingDirectory;
+                        d.CurrentDirectoryProvider = DeploymentWorkingDirectory.StagingDirectory;
                     }
-                    d.StagingDirectory = stagingDirectory;
-                    d.CurrentDirectoryProvider = DeploymentWorkingDirectory.StagingDirectory;
+
+                    kubectl.SetWorkingDirectory(d.CurrentDirectory);
+                    kubectl.SetEnvironmentVariables(d.EnvironmentVariables);
                 }),
                 new SubstituteInFilesConvention(new SubstituteInFilesBehaviour(substituteInFiles)),
-                new ConfigurationTransformsConvention(new ConfigurationTransformsBehaviour(fileSystem, variables, configurationTransformer, transformFileLocator, log)),
-                new ConfigurationVariablesConvention(new ConfigurationVariablesBehaviour(fileSystem, variables, replacer, log)),
-                new StructuredConfigurationVariablesConvention(new StructuredConfigurationVariablesBehaviour(structuredConfigVariablesService)),
+                new ConfigurationTransformsConvention(new ConfigurationTransformsBehaviour(fileSystem, variables,
+                    ConfigurationTransformer.FromVariables(variables, log),
+                    new TransformFileLocator(fileSystem, log), log)),
+                new ConfigurationVariablesConvention(new ConfigurationVariablesBehaviour(fileSystem, variables,
+                    new ConfigurationVariablesReplacer(variables, log), log)),
+                new StructuredConfigurationVariablesConvention(
+                    new StructuredConfigurationVariablesBehaviour(structuredConfigVariablesService))
             };
 
             if (variables.Get(Deployment.SpecialVariables.Account.AccountType) == "AmazonWebServicesAccount")
             {
-                conventions.Add(awsAuthConventionFactory.Create());
+                conventions.Add(new AwsAuthConvention(log, variables));
```

Comment:
I've added precompiler flags to this whole class so it won't be compiled in .NET 4.0 this means we can new up the `AwsAuthConvention` here without factories etc etc.